external/jemalloc/CMakeLists.txt

```diff
@@ -33,13 +33,21 @@ endif()
 include(ProcessorCount)
 ProcessorCount(N)
 
+set(CONFIGURE_COMMAND
+    env
+    "$<IF:$<OR:$<NOT:$<BOOL:${DESKTOP_APP_USE_PACKAGED}>>,$<BOOL:${DESKTOP_APP_SPECIAL_TARGET}>>,EXTRA_CFLAGS=-fno-omit-frame-pointer -fstack-protector-all -fstack-clash-protection -fcf-protection -DNDEBUG -D_FORTIFY_SOURCE=3 -D_GLIBCXX_ASSERTIONS,=>"
```

Comment:
What do you think about creating a CMake function like set_compile_options_if_exist to set the value of a variable for later use in EXTRA_CFLAGS? I can't use target_compile_options on a variable, it says the target doesn't exist.

Reply:
I'm not sure how to proceed here yet. But I don't really like this.

Reply:
It gets too much complicated, across... five? places

Reply:
Maybe aarch64 doesn't really worth the support given the effort...

Reply:
I had to delete the repositories to start again and don't leave rubbish because I commit the mistake to mix my personal repo with the official's fork.

If you want I can do another PR with the changes your suggested.

Here you have the commits by if you want to view them:

https://github.com/looptc/tdesktop/commit/9a717120b8d4611222ad9e420a328cba977b4b53

https://github.com/looptc/cmake_helpers/commit/004e68ea18aa8c2e18ad984a3a6b92d73587157a

Reply:
And you can use `git remote` to add the upstream remote to rebase on top of the current master branch

Reply:
Okay, thank you, I have in mind.

Reply:
The flags should remain only in Dockerfile after #357

Reply:
So, replacing the -fcf-protection by -mbranch-protection in Dockerfile will make it to work (haven't tested yet). But do you suggest some way to detect the arch or the compiler supported flags to make it automatically? To be honest, with this change I only need to replace the flag manually and maybe the future changes on the Dockerfil won't make me to re replace the flag, so with this maybe I can go nice.

Reply:
Yeah, although I wonder why [Freedesktop SDK](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/master/include/flags.yml) doesn't use -mbranch-protection?

Reply:
I don't know but security measures to face ROP are very importand by if stack canary is disabled, and even this way in some scenarios where fork() is present, can be bypassed. That's why it's important. Same question about, why distro developers doesn't give importance to binary signing and prefer to give support. I think it is because people prefers support rather than security.

Reply:
I mean they use -fcf-protection but not -mbranch-protection

Reply:
I don't know but, but is the equivalent.
https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html#index-fcf-protection
https://lists.debian.org/debian-devel/2022/10/msg00315.html
And it won't compile.
![Captura de pantalla_20240720_111352](https://github.com/user-attachments/assets/7d59f3db-c1ae-4cac-b9a1-26c648fe90cb)

Reply:
Btw, I cloned your repo tdesktop and cmake_helpers, modified the cf-protection flag by mbranch-protection=standard and it gives an segmentation fault when compiling or something glib, when running the docker run command:

FAILED: gen/external_glib_cppgir.timestamp /usr/src/tdesktop/out/gen/external_glib_cppgir.timestamp 
cd /usr/src/tdesktop/out/cmake/external/glib && /usr/src/tdesktop/out/cmake/external/glib/cppgir/Release/cppgir --debug 1 --class --class-full --expected --ignore /usr/src/tdesktop/cmake/external/glib/cppgir/data/cppgir.ignore:/usr/src/tdesktop/cmake/external/glib/cppgir/data/cppgir_unix.ignore --output /usr/src/tdesktop/out/gen Gio-2.0 && echo 1> /usr/src/tdesktop/out/gen/external_glib_cppgir.timestamp
/bin/sh: line 1:   144 Segmentation fault      (core dumped) /usr/src/tdesktop/out/cmake/external/glib/cppgir/Release/cppgir --debug 1 --class --class-full --expected --ignore /usr/src/tdesktop/cmake/external/glib/cppgir/data/cppgir.ignore:/usr/src/tdesktop/cmake/external/glib/cppgir/data/cppgir_unix.ignore --output /usr/src/tdesktop/out/gen Gio-2.0

Reply:
Perhaps it's not yet supported enough by gcc?

Reply:
It's about something related to your patches because it's compiling well with the modifications I suggested. Seems there's something broken on aarch64 with the flags from Dockerfile. If I remove mbranch-protection and cf-protection it also fails by segmentation fault.

Reply:
Well, they should provide a better debuggability, so I'm not going to remove them

Reply:
It's okay, I don't mean that. I am thankful because you helped me. I just hope this get fixed soon someway.

Reply:
It's compiling like a charm now. I followed the guide for compiling on GNU/Linux and only replaced the -cf-protection flag by -mbranch-protection=standard and that's all. Thank you all!